lib/modules/datasource/rubygems/index.ts

```diff
@@ -36,16 +35,35 @@ export class RubyGemsDatasource extends Datasource {
       /* eslint-disable @typescript-eslint/restrict-template-expressions */
       `${registryUrl}/${packageName}`,
   })
-  getReleases({
+  async getReleases({
     packageName,
     registryUrl,
   }: GetReleasesConfig): Promise<ReleaseResult | null> {
-    if (parseUrl(registryUrl)?.hostname === 'rubygems.org') {
-      return this.rubyGemsOrgDatasource.getReleases({ packageName });
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    if (!this.versionsDatasources[registryUrl]) {
+      this.versionsDatasources[registryUrl] = new VersionsDatasource(
+        RubyGemsDatasource.id,
+        registryUrl
+      );
+    }
+
+    try {
+      return await this.versionsDatasources[registryUrl].getReleases({
+        packageName,
+        registryUrl,
+      });
+    } catch (error) {
+      if (error instanceof EndpointNotSupportedError) {
```

Comment:
see other comment, you can simply check for the error message

Reply:
Updated